src/bot_methods.js

```diff
@@ -1,314 +1,498 @@
+    //logger.logMessage(message);
+
+    message.channel.send(
+      "Bot stopped! New words will not be posted automatically."
+    );
+
+    bot.clearInterval(interval);
+  } else {
+    console.log("Bot did not stop, user was not a moderator or admin");
+    //logger.logMessage(message);
+
+    message.channel.send(
+      "Bot did not stop, user must be a moderator or admin to use this command."
+    );
+  }
+}
+
+/**
+ * Handles the "help" command
+ * displays information about the bot commands in chat
+ * @param {Message} message
+ */
+function handleCommandHelp(message) {
+  console.log("Bot helped!");
+
+  let help =
+    "Commands for the word of the day bot:\n" +
+    "```" +
+    "$word x  - Searches word x in the dictionary and posts it if available\n" +
+    "$start - Starts automatic sending the word of the day - once a day, from current time (requires moderator permissions)\n" +
+    "$stop  - Stops automatic sending the word of the day (requires moderator permissions)\n" +
+    "$goodbot - Shows your appreciation for the bot\n" +
+    "$goodboy - Woof!\n" +
+    "```";
+
+  message.channel.send(help);
+}
+
+function handleChallengeMode(message) {
+  let newWord = challengeMode(message);
+  console.log(newWord[0]);
+  console.log(newWord[1]);
+  const filter = response => response.content.toLowerCase() == newWord[1].toLowerCase() || response.content.toLowerCase() == 'stop' || response.content.toLowerCase() == 'next';
+  message.channel.send("Translate: " + newWord[0]).then(() => {
+    message.channel.awaitMessages(filter, { maxMatches: 1, time: 15000, errors: ['time'] })
+      .then(collected => {
+        console.log(collected.first().content);
+        if (collected.first().content.toLowerCase() == 'stop') {
+          message.channel.send('All done!');
+          return;
+        } else if (collected.first().content.toLowerCase() == 'next') {
+          message.channel.send('The answer is: **' + newWord[1] + '**');
+          handleChallengeMode(message);
+        } else {
+          message.channel.send(`${collected.first().author} got the correct answer!`);
+          handleChallengeMode(message);
+        }
+      })
+      .catch(collected => {
+        message.channel.send('The answer I have is: **' + newWord[1] + "**");
+        handleChallengeMode(message);
+      });
+  });
+}
+
+/**
+ * handles the given command and executes the bot behaviour
+ * @param {Message} message discord message object
+ */
+function handleCommands(message) {
+  let command = getCommandFromMessage(message); // get command string
+  console.log(command);
+
+  switch (command) {
+    case "random":// user entered $random, send a random word
+      sendRandomWord(message);
+      break;
+    case "wotd": // user entered command "$wotd", send the word of the day in chat
+      //handleCommandWotd(message);
+      break;
+    case "wordf": // user entered command "$wordf x", get the word from the dictionary and send it

```

Comment:
i wanted to see whether i fixed the homonym bug, and it was hard telling things apart with both bots outputting - this way it would only make my local code run. intended to change it back after i was done

Reply:
Aight